catalog-rest-service/src/main/resources/json/schema/metadataIngestion/databaseServiceMetadataPipeline.json

```diff
@@ -3,6 +3,114 @@
   "$schema": "http://json-schema.org/draft-07/schema#",
   "title": "DatabaseServiceMetadataPipeline",
   "description": "DatabaseService Metadata Pipeline Configuration.",
+  "definitions": {
+    "credentials": {
+      "description": "GCS Credentials.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Google Cloud service account type.",
+          "type": "string"
+        },
+        "projectId": {
+          "description": "Google Cloud project id.",
+          "type": "string"
+        },
+        "privateKeyId": {
+          "description": "Google Cloud private key id.",
+          "type": "string"
+        },
+        "privateKey": {
+          "description": "Google Cloud private key.",
+          "type": "string"
+        },
+        "clientEmail": {
+          "description": "Google Cloud email.",
+          "type": "string"
+        },
+        "clientId": {
+          "description": "Google Cloud Client ID.",
+          "type": "string"
+        },
+        "authUri": {
+          "description": "Google Cloud auth uri.",
+          "type": "string"
+        },
+        "tokenUri": {
+          "description": "Google Cloud token uri.",
+          "type": "string"
+        },
+        "authProviderX509CertUrl": {
+          "description": "Google Cloud auth provider certificate.",
+          "type": "string"
+        },
+        "clientX509CertUrl": {
+          "description": "Google Cloud client certificate uri.",
+          "type": "string"
+        }
+      },
+      "additionalProperties": false
+    },
+    "credentialsPath": {
+      "description": "GCS Credentials Path.",
+      "type": "string"
+    },
+    "localHttpDBTConfig": {
+      "description": "Local and HTTP DBT configs.",
+      "type": "object",
+      "properties": {
+        "dbtCatalogFilePath": {
+          "description": "DBT catalog file to extract dbt models with their column schemas.",
+          "type": "string"
+        },
+        "dbtManifestFilePath": {
+          "description": "DBT manifest file path to extract dbt models and associate with tables.",
+          "type": "string"
+        }
+      },
+      "additionalProperties": false,
+      "required": ["dbtCatalogFilePath", "dbtManifestFilePath"]
+    },
+    "gcsDBTConfig": {
+      "description": "GCS DBT configs.",
+      "oneOf": [
+        {
+          "$ref": "#/definitions/credentials"
+        },
+        {
+          "$ref": "#/definitions/credentialsPath"
+        }
+      ]
+    },
+    "s3DBTConfig": {
```

Comment:
Lets create an S3 credentials file (plus the GCS credentials file) in the credentials path

Reply:
When refactoring this, use S3Credentials for the name or something like that, let's not specify anything about DBT

Reply:
okay I'll create separate files for S3 and GCS